feat: show version tag in process instances table

pedesen opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->

- extend processes store with computed values for a flat list of process versions and for an indicator if there are any version tags.
- add a method to the processes store which returns the version tag for a specific process id
- add version tag column to instances list, if there are any
- show version tag for each process instance (if any)

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #21327

pedesen commented:
@gastonpillet01: Please review using this image: `registry.camunda.cloud/team-operate/camunda-operate:branch-fe-21327-version-tag-instances-table`.

Please note that in the test data "Complex Process" is the only process which has a version tag named "MyVersionTag".

gastonpillet01 reviewed (commented):
I see:
- that Version tag column is always present not only for Complex process, if I select others process that hasn't version tag defined, column is being displayed with empty -- value
![image](https://github.com/user-attachments/assets/b8049005-643a-40eb-8c8a-54f7f478955b)

- in the designs it it was defined that if version tag is not defined Operate UI will not show Version tag column
![image](https://github.com/user-attachments/assets/fe196d35-279a-4c08-9ecc-54b09fba1956)

In that way it will be consistent with the header behaviour about not showing version tag if it isn't defined.

pedesen commented:
@gastonpillet01 Valid points, I will fix them.

pedesen commented on operate/client/src/App/Processes/ListView/InstancesTable/index.tsx:
```diff
@@ -47,6 +47,13 @@ const InstancesTable: React.FC = observer(() => {
+  /**
+   * Is true if at least one process instances from the store has a version tag.
+   */
+  const hasVersionTags = processInstances.some(({processId}) => {
+    return processesStore.getVersionTag(processId);
+  });
```
Now I'm checking here if there is at least one process instance in the store with a related version. If there is none found, the version tag column header won't be shown.

pedesen commented:
Hey @vitorwtavares could you please review an [additional commit](https://github.com/camunda/camunda/pull/21568/commits/4aeb8f2694e93f5a4ab9782db0997dc28d83c353) I added? based on Gaston's feedback I changed the behavior of determining whether the version tag column header is shown or not.

pedesen commented:
@gastonpillet01 I updated the behavior. Now the version tag column is only shown if there is actually at least one process instance with a version tag in the list.

gastonpillet01 reviewed (approved):
Version tag is correctly presented in process instances table.
